src/__tests__/buildTransfers.test.ts

```diff
@@ -0,0 +1,188 @@
+      );
+
+      expect(nativeTransfer.value).to.be.equal(
+        max_amount.multipliedBy(TEN.pow(18)).toFixed()
+      );
+      expect(nativeTransfer.to).to.be.equal(receiverAddress);
+      expect(nativeTransfer.data).to.be.equal("0x");
+    });
+  });
+
+  describe("Decimals", () => {
+    it("works with decimal payments on listed, unlisted and native transfers", () => {
+      let fractional_amount = new BigNumber("0.0000001");
+      let small_payments = listedUnlistedAndNativePayments(
+        fractional_amount,
+        receiverAddress
+      );
+      let [listed, unlisted, native] = buildTransfers(
+        dummySafeInfo,
+        small_payments,
+        tokenList
+      );
+      expect(listed.value).to.be.equal("0");
+      expect(listed.to).to.be.equal(listedToken.address);
+      expect(listed.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000174876e800"
+      );
+
+      expect(unlisted.value).to.be.equal("0");
+      expect(unlisted.to).to.be.equal(unlistedToken.address);
+      expect(unlisted.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000174876e800"
+      );
+
+      expect(native.value).to.be.equal(
+        fractional_amount.multipliedBy(TEN.pow(18)).toFixed()
+      );
+      expect(native.to).to.be.equal(receiverAddress);
+      expect(native.data).to.be.equal("0x");
+    });
+  });
+
+  describe("Mixed", () => {
+    it("works with arbitrary amount strings on listed, unlisted and native transfers", () => {
+      let amount = new BigNumber("123456.000000789");
+      let payments = listedUnlistedAndNativePayments(amount, receiverAddress);
+      let [listed, unlisted, native] = buildTransfers(
+        dummySafeInfo,
+        payments,
+        tokenList
+      );
+      expect(listed.value).to.be.equal("0");
+      expect(listed.to).to.be.equal(listedToken.address);
+      expect(listed.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000001a24902beecbd5109200"
+      );
+
+      expect(unlisted.value).to.be.equal("0");
+      expect(unlisted.to).to.be.equal(unlistedToken.address);
+      expect(unlisted.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000001a24902beecbd5109200"
+      );
+
+      expect(native.value).to.be.equal(
+        amount.multipliedBy(TEN.pow(18)).toFixed()
+      );
+      expect(native.to).to.be.equal(receiverAddress);
+      expect(native.data).to.be.equal("0x");
+    });
+  });
+
+  describe("Truncation on too many decimals", () => {
+    it("works with arbitrary amount strings on listed, unlisted and native transfers", () => {
+      let amount = new BigNumber("1.000000789");
```

Comment:
I think this test case only includes the unlisted case because crappy token is not on the tokenlist.
You could easily add native and listed tokens though.

Reply:
Right of course. Good point.

Reply:
I think I just forgot to rename the `it`